src/ui/public/state_management/__tests__/monitor_changes.js

```diff
@@ -0,0 +1,107 @@
+import expect from 'expect.js';
+import sinon from 'sinon';
+import monitorStateChanges from 'ui/state_management/monitor_changes';
+import SimpleEmitter from 'ui/utils/SimpleEmitter';
+
+describe('monitorStateChanges', function () {
+  const noop = () => {};
+  const eventTypes = [
+    'save_with_changes',
+    'reset_with_changes',
+    'fetch_with_changes',
+  ];
+  let mockState;
+
+  beforeEach(() => {
+    mockState = new SimpleEmitter();
+  });
+
+  describe('sanity checks', function () {
+    it('should throw if not given a handler function', function () {
+      const fn = () => monitorStateChanges(mockState, 'not a function');
+      expect(fn).to.throwException(/must be a listener function/);
+    });
+
+    it('should throw if not given a valid cleanup function', function () {
```

Comment:
Other than that's how my snippets are set up, no. Also no reason not to use `function` either though.